mllib/src/main/scala/org/apache/spark/ml/nlp/lbfgs.scala

```diff
@@ -0,0 +1,281 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.ml.nlp
+
+import scala.collection.mutable.ArrayBuffer
+
+private[ml] class Lbfgs {
```

Comment:
Spark already has a implementation of LBFGS. `org.apache.spark.mllib.optimization.LBFGS`. 
Can you apply this class into your implementation? The interface is a little bit different but `Gradient` can calculate if label and feature are included in given data. `ANNGradient` may be good resource to learn this method.